### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by JerryBerry12** [glaretechnologies/substrata#6]
Hi,
curious if there is any intention to add an ability for your avatar to sit on a bench without having to maneuver manually.
This would be kind of like the bike and hovercar scripts, but instead of having the ability to move the vehicle, it would just be the part where it positions you somewhere on the object automatically. Would be also cool to have the ability for many people to sit on one object, thus the object scripter would set the positions of where the first person would sit on the object, then the second, and so on.
IDK, just thought this would be cool to have. 
Thanks

**Comment by Ono-Sendai** [comment]
Yeah this would be cool.  There are a lot of different ways to implement this.  I think the best way might be to automatically detect any 'sittable' shapes, and have some UI for sitting on them after they are auto-detected.

**Comment by JerryBerry12** [comment]
that would probably be the most user friendly way. Another way is to just have the object creator specify (e.g. check a box) that it's sitable, then give coordinates (similar to a bike) as to where to place the user on the object when they sit.

**Comment by JerryBerry12** [comment]
bumping this. What is classifies as a "sitable object" and how would we figure that out? Possibly scanning the world every couple of minutes through all of the objects?

**Comment by Ono-Sendai** [comment]
Dunno, you tell me.  What kind of objects can humans sit on?

**Comment by JerryBerry12** [comment]
Like chairs and stuff? IDK. We could probably do it similar to the bike xml, where the object owner specifies the coordinates of the leg position, head position, etc. That would probably be more doable. If we were to do it automatically, we would probably have to implement a whole AI and stuff just for detecting what's sitable and where to place the user, and train it etc.

**Comment by JerryBerry12** [comment]
Hey, I see some cool new features in the source! (cars and stuff) Can you provide the download link for the executable for 1.6.1? I don't see anything on your website.

**Comment by Ono-Sendai** [comment]
There's a 1.6.1 build on the discord now.

**Comment by JerryBerry12** [comment]
ty!

**Comment by JerryBerry12** [comment]
Following up on this. Would the relevant code be in the Scripting.cpp file? Should probably add something there under the hovercars and bikes and boats to make this work.

**Comment by Ono-Sendai** [comment]
I'm not 100% sure where or how this would work.  You could give it a go though, trying to add some kind of <sittable> script or something,

**Comment by JerryBerry12** [comment]
Ok I'll look into it
